api/controllers/apps.go

```diff
@@ -100,6 +100,23 @@ func AppDelete(rw http.ResponseWriter, r *http.Request) *httperr.Error {
 		return httperr.Errorf(404, "invalid app: %s", name)
 	}
 
+	// check to see if the app is linked to a service, if it is fail the request with precondition failed
+	services, err := models.ListServices()
+	if err != nil {
+		return httperr.Server(err)
+	}
+	for _, serviceName := range services {
+		service, err := provider.ServiceGet(serviceName.Name)
+		if err != nil {
+			return httperr.Server(err)
+		}
+		for _, linkedApp := range service.Apps {
+			if linkedApp.Name == app.Name {
+				return httperr.Errorf(412, "%s is linked to the service %s. You must delete all service links before deleting the app %s", app, service.Name, app)
```

Comment:
That's a great point. I imagine over time it would become commonplace for apps to have multiple linked services. I'll defer to @ddollar for the preferred user experience but populating an array and returning that won't be difficult.